lifetimes/utils.py

```diff
@@ -114,6 +115,32 @@ def to_period(d):
     return customers[['frequency', 'recency', 'T']].astype(float)
 
 
+def calculate_alive_path(model, transactions, datetime_col, units, freq='D'):
+    """
+    :param model: A fitted lifetimes model
+    :param transactions: a Pandas DataFrame containing the transactions history of the customer_id
+    :param datetime_col: the column in the transactions that denotes the datetime the purchase was made
+    :param units: the number of time units since the birth we want to draw the p_alive
+    :param freq: Default 'D' for days. Other examples= 'W' for weekly
+    :return: A pandas Series containing the p_alive as a function of T (age of the customer)
+    """
+    _df = pd.DataFrame(transactions[datetime_col].copy())
+    _df.index = _df[datetime_col]
+    # Add transactions column
+    _df['transactions'] = 1
+    purchase_history = _df.resample(freq, how='sum').reset_index().fillna(0).drop([datetime_col], axis=1)['transactions'].values
+    extra_columns = units - len(purchase_history)
+    _df = pd.DataFrame(np.append(purchase_history, [0] * extra_columns), columns=['transactions'])
+    # add T column
+    _df['T'] = _df.index.values
+    # add cumulative transactions column
+    _df['x_sum'] = _df['transactions'].cumsum() - 1  # first purchase is ignored
+    # Add t_x column
+    _df['t_x'] = _df.apply(lambda row: row['T'] if row['transactions'] != 0 else 0, axis=1)
+    _df['t_x'] = _df['t_x'].replace(0, np.nan).fillna(method='ffill').fillna(0)
```

Comment:
these two lines are clever. What about 
```
_df['t_x'] = _df.apply(lambda row: row['T'] if row['transactions'] != 0 else np.nan, axis=1)
```
so you don't need the later `replace`.